clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinatorTest.java

```diff
@@ -466,24 +477,193 @@ public void testSyncGroupRequestWithFencedInstanceIdException() {
     }
 
     @Test
-    public void testHeartbeatUnknownMemberResponseDuringRebalancing() throws InterruptedException {
+    public void testJoinGroupUnknownMemberResponseWithOldGeneration() throws InterruptedException {
         setupCoordinator();
-        mockClient.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
+        joinGroup();
 
-        final int generation = 1;
+        final AbstractCoordinator.Generation currGen = coordinator.generation();
 
-        mockClient.prepareResponse(joinGroupFollowerResponse(generation, memberId, JoinGroupRequest.UNKNOWN_MEMBER_ID, Errors.NONE));
-        mockClient.prepareResponse(syncGroupResponse(Errors.NONE));
+        RequestFuture<ByteBuffer> future = coordinator.sendJoinGroupRequest();
 
-        coordinator.ensureActiveGroup();
+        TestUtils.waitForCondition(() -> !mockClient.requests().isEmpty(), 2000,
+            "The join-group request was not sent in time after");
```

Comment:
nit: after.. what? I think you can drop "in time after." Here is the assertion that is used:
```
            assertThat("Condition not met within timeout " + maxWaitMs + ". " + conditionDetails,
                testCondition.conditionMet());
```

Reply:
Ack.